src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -580,30 +583,29 @@ def _search_role_assignments(cli_ctx, assignments_client, definitions_client,
     # "atScope()" and "principalId eq '{value}'" query cannot be used together (API limitation).
     # always use "scope" if provided, so we can get assignments beyond subscription e.g. management groups
     if scope:
-        f = 'atScope()'  # atScope() excludes role assignments at subscopes
-        if assignee_object_id and include_groups:
-            f = f + " and assignedTo('{}')".format(assignee_object_id)
-        assignments = list(assignments_client.list_for_scope(scope=scope, filter=f))
-    elif assignee_object_id:
-        if include_groups:
-            f = "assignedTo('{}')".format(assignee_object_id)
+        filters = []
+        if at_scope:
+            filters.append('atScope()')  # atScope() excludes role assignments at subscopes
+            if assignee_object_id and include_groups:
+                filters.append("assignedTo('{}')".format(assignee_object_id))
         else:
-            f = "principalId eq '{}'".format(assignee_object_id)
-        assignments = list(assignments_client.list_for_subscription(filter=f))
```

Comment:
There is no need to call [`list_for_subscription`](https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/list-for-subscription?view=rest-authorization-2022-04-01&tabs=HTTP) at all. Its functionality is already covered by [`list_for_scope`](https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/list-for-scope?view=rest-authorization-2022-04-01&tabs=HTTP).